src/routing_table/mod.rs

```diff
@@ -502,6 +502,11 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
     // The actual merge of the group is only done once all expected merging groups have provided
     // details.  See the docs for `OwnMergeState` for full details of the return value.
     pub fn merge_own_group(&mut self, merge_details: OwnMergeDetails<T>) -> OwnMergeState<T> {
+        if self.groups.contains_key(&merge_details.merge_prefix) {
+            warn!("{:?}: Attempt to call merge_own_group() for an already merged prefix {:?}",
+                  self.our_name.debug_binary(),
```

Comment:
It was more convenient for debugging some issues this way (don't remember the details, unfortunately, but it was easier to compare the node from the message with RT printouts), but I can of course change it.